include/psa/crypto_values.h

```diff
@@ -1814,10 +1814,21 @@
  * The latter is based on HMAC and can be used with either SHA-256
  * or SHA-384.
  *
+ * In a mixed-PSK handshake (RSA-PSK, DHE-PSK, ECDHE-PSK) in TLS 1.2,
+ * the PSK is concatenated with the output of RSA key wrapping or
+ * of (EC)DHE key agreement to form the pre-master secret (PMS)
+ * from which the MS is derived. For details please check:
+ * - RFC 4279, Section 4 for the definition of RSA-PSK,
+ * - RFC 4279, Section 3 for the definition of DHE-PSK,
+ * - RFC 5489 for the definition of ECDHE-PSK.
+ *
  * This key derivation algorithm uses the following inputs, which must be
  * passed in the order given here:
  * - #PSA_KEY_DERIVATION_INPUT_SEED is the seed.
- * - #PSA_KEY_DERIVATION_INPUT_SECRET is the secret key.
+ * - #PSA_KEY_DERIVATION_INPUT_SALT is the the output of RSA
```

Comment:
I felt conflicted about this too. Thanks for the point about key type, I had failed to consider that. I think this makes me more inclined to use a new, dedicated input type, then, if input type proliferation is not considered an issue.

Reply:
Reading all the comments about how other secret input should be provided (using existing `SALT` input vs new `OTHER_SECRET` input) I'm getting confused.

> On the gripping hand, using SALT requires a change to the specification of PSA_KEY_DERIVATION_INPUT_SALT, because when using input_key, that requires an input of type RAW_DATA or PEPPER. But for TLS12_PSK_TO_MS, the input type should be DERIVE instead. (Please ensure there are tests for this.) Arguably, SALT might permit DERIVE inputs for any KDF, but I fear confusion between inputs for other KDF then.

https://github.com/Mbed-TLS/mbedtls/pull/5685/files#r844122587

Currently the PSK is provided using `psa_key_derivation_input_key()` where where key needs to be of DERIVE type (nothing changes here).
Other key is provided as salt using `psa_key_derivation_input_bytes()`, so other key is provided as raw data.
I don't see the issue here.

Reply:
@mprse I'm not sure if you have access to the [PSA PR](https://github.com/ARM-software/psa-crypto-api/pull/549) but in case you don't: the discussion there seems to have converged to using a new type called `PSA_KEY_DERIVATION_INPUT_OTHER_SECRET`.

Reply:
> Currently the PSK is provided using psa_key_derivation_input_key() where where key needs to be of DERIVE type (nothing changes here).
Other key is provided as salt using psa_key_derivation_input_bytes(), so other key is provided as raw data.

All inputs can be provided either with `input_key` or with `input_bytes`, at the application's choice. With `input_key`, there are [constraints of type](https://armmbed.github.io/mbed-crypto/html/api/ops/kdf.html?highlight=type#c.PSA_KEY_DERIVATION_INPUT_SALT). Some inputs can be provided with `key_agreement` (https://armmbed.github.io/mbed-crypto/html/api/ops/kdf.html?highlight=psa_key_derivation_key_agreement#c.PSA_KEY_DERIVATION_INPUT_SECRET) as a third option.

Reply:
In the future we want to use `input_key` in a lot more places, included here. That's Goal 3 "Allow isolation of short-term secrets (for example, TLS session keys)." in the [strategy document](https://github.com/Mbed-TLS/mbedtls/blob/development/docs/architecture/psa-migration/strategy.md). We're no yet to the point where doing it is in the next step, but we should make sure we're not blocking that path.

Reply:
Yes, I have access, so the design under PR https://github.com/ARM-software/psa-crypto-api/pull/549 is valid and I need to adapt implementation to it?

Reply:
I think so, unless we're expecting more review on the psa-crypto-api PR - @gilles-peskine-arm ?

Reply:
Any further review would likely be from the Mbed TLS team — probably you two.

So @mprse please follow https://github.com/ARM-software/psa-crypto-api/pull/549, but feel free to request changes there if you think something isn't right. Since you're the first to implement it, you may well discover that what I wrote on paper doesn't work in practice, or isn't precise enough.